d2layouts/d2sequence/sequence_diagram.go

```diff
@@ -63,21 +101,34 @@ func newSequenceDiagram(actors []*d2graph.Object, messages []*d2graph.Edge) *seq
 		queue := make([]*d2graph.Object, len(actor.ChildrenArray))
 		copy(queue, actor.ChildrenArray)
 		for len(queue) > 0 {
-			span := queue[0]
+			child := queue[0]
 			queue = queue[1:]
 
-			// spans are always rectangles and have no labels
-			span.Attributes.Label = d2graph.Scalar{Value: ""}
-			span.Attributes.Shape = d2graph.Scalar{Value: shape.SQUARE_TYPE}
-			sd.spans = append(sd.spans, span)
-			sd.objectRank[span] = rank
-
-			queue = append(queue, span.ChildrenArray...)
+			// spans are children of actors that have edges
+			// notes are children of actors with no edges
+			if hasEdge(child) {
+				// spans have no labels
+				// TODO why not? Spans should be able to
+				child.Attributes.Label = d2graph.Scalar{Value: ""}
+				child.Attributes.Shape = d2graph.Scalar{Value: shape.SQUARE_TYPE}
+				sd.spans = append(sd.spans, child)
+				sd.objectRank[child] = rank
+			} else {
+				sd.verticalIndices[child.AbsID()] = getObjEarliestLineNum(child)
```

Comment:
what about deeper descendants.
It seems that they would be drawn as notes too

Reply:
i thought about this.

when we have "edge groups", notes should be able to be descendants of them.

for spans, there will be an edge reference.